src/consensus/encode.rs

```diff
@@ -306,6 +308,37 @@ pub trait Encodable {
     ///
     /// The only errors returned are errors propagated from the writer.
     fn consensus_encode<W: io::Write + ?Sized>(&self, writer: &mut W) -> Result<usize, io::Error>;
+
+    /// Objects with statically known serialized len, such as [`crate::Txid`] or [`crate::BlockHeader`],
+    /// have the serialized len value here, 0 means the object size is known only at run-time, for 
+    /// example contains a `Vec`.
+    const STATIC_SERIALIZED_LEN: usize = 0;
+
+    /// Returns the bytes needed when `self` is serialized
+    ///
+    /// The computation could be costly for big objects, see [`Encodable::serialized_len_early_stop`] to limit
+    /// the cost at the expense of precision for big objects
+    fn serialized_len(&self) -> usize {
+        if Self::STATIC_SERIALIZED_LEN == 0 {
+            self.consensus_encode(&mut sink()).unwrap()
```

Comment:
Regrettably this will never go back to checking `STATIC_SERIALIZED_LEN`/and using specialized versions of this function on any inner part of `self` anymore, which makes implementing specialized version of this in all larger object important. 


Possibly it would be better to just bite the bullet and just not have default implementation for these functions at all, to just force optimal implementation.

Reply:
Yeah, I'd also prefer to not provide defaults.

Reply:
> Possibly it would be better to just bite the bullet and just not have default implementation for these functions at all, to just force optimal implementation that recursively uses serialized_len on all fields.

Strongly agree

Reply:
https://github.com/rust-bitcoin/rust-bitcoin/pull/1027 here for reference